src/roiextractors/extractors/femtonicsimagingextractor/femtonicsimagingextractor.py

```diff
@@ -54,42 +63,66 @@ def __init__(
-            raise ValueError(
-                f"Session index {self._session_index} not found. " f"Available sessions: {available_sessions}"
-            )
+    def _setup_session_selection(self):
+        """Select the session to use with proper validation and error handling."""
+        available_sessions = self._get_available_sessions_from_file()
 
-        # Get available units in this session
-        session_group = self._file_handle[self._session_key]
-        available_units = [k for k in session_group.keys() if k.startswith("MUnit_")]
+        if not available_sessions:
+            raise ValueError(f"No sessions found in file: {self.file_path}")
 
-        if self._munit_index >= len(available_units):
-            raise ValueError(
-                f"Unit index {self._munit_index} not found in {self._session_key}. "
-                f"Available units: {available_units}"
-            )
+        if self._session_name is None:
+            if len(available_sessions) > 1:
+                raise ValueError(
+                    f"Multiple sessions found in file: {self.file_path}. "
+                    f"Available sessions: {available_sessions}. "
+                    "Please specify 'session_name' to select one."
+                )
+            self._selected_session_name = available_sessions[0]
+        else:
+            if self._session_name not in available_sessions:
+                raise ValueError(
+                    f"Session '{self._session_name}' not found in file: {self.file_path}. "
+                    f"Available sessions: {available_sessions}"
+                )
+            self._selected_session_name = self._session_name
+
+    def _setup_munit_selection(self):
+        """Select the MUnit to use within the selected session."""
+        available_munits = self._get_available_munits(self._selected_session_name)
 
-        # Set the actual unit key (not assuming index matches)
-        self._munit_key = available_units[self._munit_index]
+        if not available_munits:
+            raise ValueError(f"No MUnits found in session '{self._selected_session_name}' of file: {self.file_path}")
+
+        if self._munit_name is None:
+            if len(available_munits) > 1:
+                raise ValueError(
+                    f"Multiple MUnits found in session '{self._selected_session_name}' of file: {self.file_path}. "
+                    f"Available MUnits: {available_munits}. "
+                    "Please specify 'munit_name' to select one."
+                )
+            self._selected_munit_name = available_munits[0]
+        else:
+            if self._munit_name not in available_munits:
+                raise ValueError(
+                    f"MUnit '{self._munit_name}' not found in session '{self._selected_session_name}' "
+                    f"of file: {self.file_path}. Available MUnits: {available_munits}"
+                )
+            self._selected_munit_name = self._munit_name
```

Comment:
My instinct is to not create a new variable (`self._selected_m_unit_name`) and instead modify `self._m_unit_name` since both are private. 

Buy maybe this is better for provenance, how are you thinking about it?

Reply:
Ok. Let's go with it, I don't think it hurts and it is maybe a good idea.